src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -144,16 +144,22 @@ void ObjectBufferPool::AbortCreateChunk(const ObjectID &object_id,
   }
 }
 
-void ObjectBufferPool::SealChunk(const ObjectID &object_id, const uint64_t chunk_index) {
+void ObjectBufferPool::WriteChunk(const ObjectID &object_id, const uint64_t chunk_index,
+                                  const std::string &data) {
   std::lock_guard<std::mutex> lock(pool_mutex_);
   auto it = create_buffer_state_.find(object_id);
   if (it == create_buffer_state_.end() ||
-      it->second.chunk_state[chunk_index] != CreateChunkState::REFERENCED) {
+      it->second.chunk_state.at(chunk_index) != CreateChunkState::REFERENCED) {
     RAY_LOG(DEBUG) << "Object " << object_id << " aborted due to OOM before chunk "
                    << chunk_index << " could be sealed";
     return;
   }
-  it->second.chunk_state[chunk_index] = CreateChunkState::SEALED;
```

Comment:
Can you add a check here just in case (afaik, if exception is raised, it was not properly logged currently). 

```cpp
RAY_CHECK(it->second.chunk_info.size() >= chunk_index);
```